frontend/app/services/subscription-service.server.ts

```diff
@@ -242,41 +277,111 @@ function createSubscriptionService() {
     }
   }
 
-  //TODO: this one will be updated in another PR
-  async function updateSubscription(sin: string, subscription: SubscriptionInfo) {
+  async function createSubscription(userId: string, subscription: SubscriptionInfo) {
     const auditService = getAuditService();
     const instrumentationService = getInstrumentationService();
-    auditService.audit('alert-subscription.update', { sin });
+    auditService.audit('alert-subscription.update', { userId });
+    const url = new URL(`${CDCP_API_BASE_URI}/api/v1/users/${userId}`);
+    const response = await fetch(url, {
```

Comment:
I think this should use the new `instrumentedFetch()` function: <https://github.com/DTS-STN/canadian-dental-care-plan/blob/main/frontend/app/utils/fetch-utils.server.ts#L55>

Reply:
Also.. I think the user data should be stored in session at this point, shouldn't it? Do we have to create a new `fetch()` call just to get the user's subscriptions resource?

Reply:
Since we're using the HATEOAS `links` within the `GET /users` response to find the subscriptions resource, we would have to store that particular `link` within the user data in the session to use in subsequent requests to the API (unless there's another way I didn't think of?).

I feel that's odd because by doing that, the user data in the session has a mix of domain data (eg. user's email) and implementation details (eg. `link` to `subscription` resource).

Reply:
I see your point although I don't entirely agree (I don't see any problem with storing links in session).

@leiye29 ignore my comment and follow @nicholas-ly's advice.

If we're going to be making multiple HTTP calls, though, we should ensure that we have adequate fatch caching using `Cache-Control` and `ETag` headers. This is something for a future PR.